views/course_form/views.py

```diff
@@ -33,7 +171,41 @@ def delete(self, request, code: str, semester: str):
             administrator then the course with the given info is deleted from the database
         Side-effects: Course and all linked sections/assignments are deleted
         '''
-        #  Included so that tests fail instead of throwing errors
-        return render(request, 'login/login.html')
+
+        print(f"User Role: {request.user.role}")  # Debugging
+        print(f"User Authenticated: {request.user.is_authenticated}")
+
+
+        # Check if the user is authenticated
+        if not request.user.is_authenticated:
+            return HttpResponse("User not authenticated.", status=401)
+
+            # Restrict only TAs from deleting courses
+        if request.user.role == "TA":
+            print("Blocked deletion: User is a TA.")  # Debugging
+            raise PermissionDenied("TAs cannot delete courses")
```

Comment:
views shouldn't raise errors. It will just crash the site. They should handle error. In most cases right now we just redirect to home. Unsure if not covered by tests.